operator-common/src/main/java/io/strimzi/operator/common/ShutdownHook.java

```diff
@@ -38,22 +45,48 @@ public ShutdownHook(Vertx vertx, long timeoutMs) {
     @Override
     public void run() {
         LOGGER.info("Shutdown started");
-        if (vertx.deploymentIDs().size() > 0) {
-            CountDownLatch latch = new CountDownLatch(1);
-            vertx.close(ar -> {
-                if (!ar.succeeded()) {
-                    LOGGER.error("Vertx close failed", ar.cause());
-                }
-                latch.countDown();
-            });
+        vertx.deploymentIDs()
+                .stream()
+                .map(id -> ((VertxImpl) vertx).getDeployment(id))
+                .forEach(deployment -> stopVerticles(deployment));
+        LOGGER.info("Shutdown completed");
+    }
+
+    private void stopVerticles(Deployment deployment) {
+        Set<Verticle> verticles = deployment.getVerticles();
+        ExecutorService executor = Executors.newFixedThreadPool(verticles.size());
```

Comment:
Is it really the right choice to start starting new executors when you are shutting down?

Reply:
That is to speed up the shutdown in case of deployments with multiple verticles. Let me also review this one.

Reply:
I think the shutdown was completely fine before the shutdown hook. It seems really weird to now _shutdown_ by starting more new things.